meshio/vtk/_vtk.py

```diff
@@ -629,10 +629,10 @@ def pad(array):
             _write_field_data(f, mesh.point_data, binary)
 
         # write cell data
-        for name, data in mesh.cell_data.items():
-            total_num_cells = sum([len(c) for c in data])
+        if mesh.cell_data:
+            total_num_cells = sum(len(c.data) for c in mesh.cells)
             f.write(f"CELL_DATA {total_num_cells}\n".encode("utf-8"))
-            _write_field_data(f, raw_from_cell_data(mesh.cell_data), binary)
+            _write_field_data(f, mesh.cell_data, binary)
```

Comment:
Can we really just write `mesh.cell_data`? I should be a dict of lists of numpy.arrays, really. `raw_from` converts this to a dict of numpy.array by concatenation.

Reply:
Yes, `mesh.cell_data` is a dict of lists of arrays and the lists must be joined. This is done in  `_write_field_data()`, if `data` is a list. This produces just one `CELL_DATA` + `FIELD` section followed by the fields.

There was probably something strange in the commit, I didn't see the changes in `_write_field_data()`, after re-pushing, I see it again.

Reply:
Alright, since the tests pass, let's give it a go.